### DISCUSSION TITLE
module configuration: forecast days

### DISCUSSION THREAD

**Discussion by lhsans** [jeroenterheerdt/HAsmartirrigation#496] [Q&A]
Hello,

First thing first, this is a great integration for HA! many thanks to the developper(s).

I was wondering about the "forecast days" option in the module (PyETO) configuration. The docs state:
_Forecast days: How many forecast days taken into account. This is useful to avoid irrigating if the forecast suggests it will rain._

Does this mean that only forecasted rain is taken into account (ie. added to the bucket) or does the forecast also take (forecasted) Et into account.

(for context, my usecase: irrigation 2x per week (or when bucket < -25mm), maxbucket 30mm. I add some extra mm (calculated back to zone duration) to the bucket when I start the irrigation. So after watering my bucket is not 0 but +10 for example. I use this to "survive" 2-3days without irrigation, mainly usefull in summer. If the "forecast days" includes future Et values, then I should not be adding to the bucket :) )

Thanks

**Comment by Seven-709** [comment]
Hello,

I have a similar question about "forecast". My weather station is providing all the needed sensor values. I have configured one sensor group to use these sensors accordingly. In case I configure Smart Irrigation now with a weather service and enter a value > 0 for "forecast days" in the PyETO module, will the forecast values from the weather service loaded somehow in the background automatically? I didn't configure "weather service" anyhere else.

Also in this case, can I continue to use the "Continuous Updates for Sensors" feature without polling the API for the weather service?

Thanks in advance

**Comment by jeroenterheerdt** [comment]
You can only enable forecast > 0 if you've set up a weather service.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Seven-709** [reply]
Hello,
thank you very much for your quick response. To avoid a possible misunderstanding I will rephrase my question ;-)

Consider the following scenario:

- Assume that for baseline configuration, "PirateWeather" is configured as weather service with a valid API Key
- Furthermore assume that there is one, and only one, sensor group in use. However, the source for all values are entities from HomeAssistant of my local weather station. The above-mentioned weather service is not used in this sensor group.

In this scenario:
- Will the forecast days, which can be configured in the PyETO module pulled from the "PirateWeather" weather service, despite the fact that it is not used/configured in the sensor group?
- Will the enabled weather service be affected by the feature "Continous Update for sensors", although it is not used/configured in the sensor group? In the sense that API calls are triggered against PirateWeather API?

I hope I have made myself more clear now, I'm sorry for causing confusion due to my previous unclear post

thanks in advance

**Reply by jeroenterheerdt** [reply]
Will the forecast days, which can be configured in the PyETO module pulled from the "PirateWeather" weather service, despite the fact that it is not used/configured in the sensor group?

Yes as your weather station does not provide forecasts and if it did Smart Irrigation will not use them. Forecasts are retrieved from the weather service you set up, if any (PirateWeather in your example) 

Will the enabled weather service be affected by the feature "Continous Update for sensors", although it is not used/configured in the sensor group? In the sense that API calls are triggered against PirateWeather API?

No, continuous updates are for sensors only, as the name says.

**Reply by Seven-709** [reply]
Hello,

splendid, many thanks for your answers. The rest of "Smart Irrigation" integration is either self-explenatory or documented. I will now switch from from classic "manual" to smart irrigation (pun intended) for my garden step by step.

Again, many thanks for the great support!

**Comment by jeroenterheerdt** [comment]
Great! I'd welcome a PR for the docs to make this clearer!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
